modules/setting/setting.go

```diff
@@ -208,17 +205,29 @@ func PrepareAppDataPath() error {
 
 // InitProviderFromExistingFile initializes config provider from an existing config file (app.ini)
 func InitProviderFromExistingFile() {
-	CfgProvider = newFileProviderFromConf(CustomConf, false, "")
+	var err error
+	CfgProvider, err = newConfigProviderFromFile(CustomConf, false, "")
+	if err != nil {
+		log.Fatal("%v", err)
```

Comment:
Last nit, IMO it's better to add message context / prefix for logs: `log.Fatal("InitProviderFromExistingFile: %v", err)`

Otherwise if a user reports an error message, developers are not able to know where the message come from.

(and below)